lib/chef/http.rb

```diff
@@ -142,13 +142,26 @@ def delete(path, headers = {})
     # Makes an HTTP request to +path+ with the given +method+, +headers+, and
     # +data+ (if applicable).
     def request(method, path, headers = {}, data = false)
+      http_attempts ||= 0
       url = create_url(path)
       method, url, headers, data = apply_request_middleware(method, url, headers, data)
 
       response, rest_request, return_value = send_http_request(method, url, headers, data)
       response, rest_request, return_value = apply_response_middleware(response, rest_request, return_value)
+
       response.error! unless success_response?(response)
       return_value
+
+    rescue Net::HTTPServerException => e
```

Comment:
We have to do this here (and in the streaming requests) rather than in `retrying_http_errors` because we need to re-process all the middleware each time.